go/vt/vtgate/vtgateconn/vtgateconn.go

```diff
@@ -183,7 +190,10 @@ func RegisterDialer(name string, dialer DialerFunc) {
 
 // DialProtocol dials a specific protocol, and returns the *VTGateConn
 func DialProtocol(ctx context.Context, protocol string, address string) (*VTGateConn, error) {
+	dialersM.Lock()
 	dialer, ok := dialers[protocol]
+	dialersM.Unlock()
```

Comment:
Is there a reason for not using `defer` beyond "it's like, one line"? (Purely curious/not a leading question lol.)

Reply:
because we don't need to hold the mutex to invoke the dialer, just to get it out of the map